utils/estimated_histogram.hh

```diff
@@ -47,13 +47,376 @@
 #include <chrono>
 #include <seastar/core/metrics_types.hh>
 #include <seastar/core/print.hh>
-
-#include <seastar/core/print.hh>
-
 #include "seastarx.hh"
+#include <seastar/core/bitops.hh>
+#include <limits>
 
 namespace utils {
 
+
+// Returns largest N such that 2^N <= v.
+// Undefined for v == 0.
+inline size_t pow2_rank(uint64_t v) {
+    return std::numeric_limits<uint64_t>::digits - 1 - count_leading_zeros(v);
+}
+
+
+// Returns largest N such that 2^N <= v.
+// Undefined for v == 0.
+inline constexpr size_t pow2_rank_constexpr(uint64_t v) {
+    return v <= 1 ? 0 : 1 + pow2_rank_constexpr(v >> 1);
+}
+
+template<uint64_t MIN, uint64_t MAX, std::vector<uint64_t>::size_type NUM_BUCKETS>
+struct pseudo_floatingpoint_histogram {
```

Comment:
I Used template parameters to help the compiler optimize, technically I can even damp the vector and replace it with C array because of that (it brought only an additional 10% improvement so I didn't)